src/api/app/views/webui2/webui/project/_range_month.html.haml

```diff
@@ -0,0 +1 @@
+= "#{Date.today.strftime("%B, %e")} – #{Date.today.prev_month.strftime("%B, %e")}"
```

Comment:
This belongs to a helper

Reply:
Can't do this, I have to render it in `update_pulse.js.erb`...

Reply:
Because the `update_pulse.js.erb` view renders the partial and replaces some html in the `show` view with it.

Reply:
This as well:
```diff
diff --git a/src/api/app/views/webui2/webui/project/update_pulse.js.erb b/src/api/app/views/webui2/webui/project/update_pulse.js.erb
index b9c019523d..d211c9680c 100644
--- a/src/api/app/views/webui2/webui/project/update_pulse.js.erb
+++ b/src/api/app/views/webui2/webui/project/update_pulse.js.erb
@@ -1,3 +1,3 @@
-$('#range-header').html("<%= escape_javascript render partial: "range_#{@range}" %>");
+$('#range-header').html("<%= escape_javascript(pulse_period(@range)) %>");
diff --git a/src/api/app/helpers/webui/project_helper.rb b/src/api/app/helpers/webui/project_helper.rb
index a7ed2b0b95..37b96b4633 100644
--- a/src/api/app/helpers/webui/project_helper.rb
+++ b/src/api/app/helpers/webui/project_helper.rb
@@ -3,6 +3,16 @@ module Webui::ProjectHelper
 
   protected
 
+  def pulse_period(range)
+    start = if range == 'month'
+              Time.zone.today.prev_month.strftime('%B, %e')
+            else
+              Time.zone.today.prev_week.strftime('%B, %e')
+            end
+
+    "#{Time.zone.today.strftime('%B, %e')} – #{start}"
+  end

```
Since I already wrote some code to try it out, I created a PR:
https://github.com/hennevogel/open-build-service/pull/8/commits/7123cf8167e386c8e56fe0c97bcab5fe4167cc73